cmd/snap-gdb-shim/snap-gdb-shim.c

```diff
@@ -15,20 +15,64 @@
  *
  */
 
+#include <errno.h>
 #include <signal.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <unistd.h>
 
 #include "../libsnap-confine-private/utils.h"
 
+int must_parse_int(const char *s)
+{
+	char *endptr = NULL;
+
+	errno = 0;
+	long i = strtol(s, &endptr, 10);
+	if (errno != 0 || s == endptr || *endptr != '\0' || i < 0) {
+		die("cannot parse number in '%s'", s);
+	}
+	return i;
+}
+
 int main(int argc, char **argv)
 {
 	if (sc_is_debug_enabled()) {
 		for (int i = 0; i < argc; i++) {
 			printf("-%s-\n", argv[i]);
 		}
 	}
+
```

Comment:
As a style suggestion I would unconditionally load the environment variable here:
```
const char *sudo_uid_env = getenv("SUDO_UID");
```

And then merge the if condition into one, to avoid one level of nesting:
```
if (sudo_uid_env != NULL && getuid() == 0) {
   ...
}
```